src/merkle_tree.cairo

```diff
@@ -1,88 +1,355 @@
-use super::utils::double_sha256;
+use super::utils::{double_sha256, Hash};
 
-pub fn merkle_root(ref txids: Array<u256>) -> u256 {
-    let len = txids.len();
+pub fn merkle_root(ref hashes: Array<Hash>) -> Hash {
+    let len = hashes.len();
 
     if len == 1 {
-        return *txids.at(0);
+        return *hashes.at(0);
```

Comment:
cleaner to use subscripting operator when accessing elements in an array

```
*hashes.at(0)
```

becomes

```
*hashes[0]
```